Mathlib/NumberTheory/GaussSum.lean

```diff
@@ -92,7 +92,7 @@ variable {R : Type u} [Field R] [Fintype R] {R' : Type v} [CommRing R'] [IsDomai
 
 -- A helper lemma for `gaussSum_mul_gaussSum_eq_card` below
 -- Is this useful enough in other contexts to be public?
-private theorem gaussSum_mul_aux {χ : MulChar R R'} (hχ : IsNontrivial χ) (ψ : AddChar R R')
+private theorem gaussSum_mul_aux {χ : MulChar R R'} (hχ : χ.IsNontrivial) (ψ : AddChar R R')
```

Comment:
This is a bug, it seems. Without the fix, Lean successfully deduces that we must mean `MulChar.IsNontrivial`, because `AddChar.IsNontrivial` would not fit the arguments; but it still emits the deprecation warning for the declaration it _didn't_ use!

We should clearly get rid of `MulChar.IsNontrivial` too and then this will be moot; but the bug is worth reporting anyway, I think.

Reply:
It's already known. See https://github.com/leanprover/lean4/issues/2132 (it might seem like a slightly different issue, but it is actually the same, as confirmed on Zulip by Mario)